.github/workflows/pull_request.yml

```diff
@@ -21,6 +21,43 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Shellcheck
+        uses: reviewdog/action-shellcheck@v1
+        with:
+          github_token: ${{ secrets.github_token }}
+          reporter: github-pr-review
+          path: "."
+          pattern: "*.sh"
+
+      - uses: cachix/install-nix-action@v22
+      - uses: rrbutani/use-nix-shell-action@v1.1.0
+        with:
+          file: shell.nix
+
+      - name: Set up Go 1.19
+        uses: actions/setup-go@v5
+        with:
+          go-version: 1.19.13
+          cache: false
+
+      - name: Lint Check
+        uses: golangci/golangci-lint-action@v4
+        with:
+          version: v1.56.2
+          args: -E exportloopref,dupl,revive,bodyclose,goconst,misspell -D structcheck --timeout 5m0s
```

Comment:
@niladrih Should we remove golangci-lint? I see lot of failures. Unless you want to fix them :-p

Reply:
Let's ignore them for now? Let's leave them failing and fix the failures in a different PR..IMO

Reply:
So for merging should I disable the check or force push ?

Reply:
Commented out the golang ci lint for now.